src/StepperFirmata2.cpp

```diff
@@ -0,0 +1,401 @@
+        } else if (wireCount == 3 && stepType == 0) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::FULL3WIRE, directionPin, stepPin, motorPin3);
+        } else if (wireCount == 3 && stepType == 2) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::HALF3WIRE, directionPin, stepPin, motorPin3);
+        } else if (wireCount == 4 && stepType == 0) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::FULL4WIRE, directionPin, stepPin, motorPin3, motorPin4, false);
+        } else if (wireCount == 4 && stepType == 2) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::HALF4WIRE, directionPin, stepPin, motorPin3, motorPin4);
+        }
+
+        if (argc >= index) {
+          invertPins = argv[index]; // optional byte 11, 12 or 13
+          if (wireCount == 0) {
+             stepper[deviceNum]->setPinsInverted(invertPins & 0x01, invertPins >> 1 & 0x01, invertPins >> 4 & 0x01);
+          } else {
+            stepper[deviceNum]->setPinsInverted(invertPins & 0x01, invertPins >> 1 & 0x01, invertPins >> 2 & 0x01, invertPins >> 3 & 0x01, invertPins >> 4 & 0x01);
+          }
+        }
+
+        if (interface & 0x01) {
+          stepper[deviceNum]->setEnablePin(enablePin);
+        }
+
+        isRunning[deviceNum] = false;
+
+      }
+
+      else if (stepCommand == STEPPER_STEP) {
+
+        numSteps = decode32BitSignedInteger(argv[2], argv[3], argv[4], argv[5], argv[6]);
+
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->move(numSteps);
+          isRunning[deviceNum] = true;
+        }
+
+      }
+
+      else if (stepCommand == STEPPER_ZERO) {
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->setCurrentPosition(0);
+        }
+      }
+
+      else if (stepCommand == STEPPER_TO) {
+        if (stepper[deviceNum]) {
+          numSteps = decode32BitSignedInteger(argv[2], argv[3], argv[4], argv[5], argv[6]);
+          stepper[deviceNum]->moveTo(numSteps);
+          isRunning[deviceNum] = true;
+        }
+      }
+
+      else if (stepCommand == STEPPER_ENABLE) {
+        if (stepper[deviceNum]) {
+          if (argv[2] == 0x00) {
+            stepper[deviceNum]->disableOutputs();
+          } else {
+            stepper[deviceNum]->enableOutputs();
+          }
+        }
+      }
+
+      else if (stepCommand == STEPPER_STOP) {
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->stop();
+          isRunning[deviceNum] = false;
+          reportPosition(deviceNum, true);
+        }
+      }
+
+      else if (stepCommand == STEPPER_REPORT_POSITION) {
+        if (stepper[deviceNum]) {
+          reportPosition(deviceNum, false);
+        }
+      }
+
+      else if (stepCommand == STEPPER_SET_ACCELERATION) {
+
+        float decodedAcceleration = decodeCustomFloat(argv[2], argv[3], argv[4], argv[5]);
+
+        if (stepper[deviceNum]) {
+          if (decodedAcceleration == 0.0) {
+            stepper[deviceNum]->setAcceleration(99999);
```

Comment:
I had run through the run vs runSpeed options and even had an implementation where it chose which to call based on the current acceleration value for that particular stepper. I was having to track the acceleration value in firmata since it cannot be read from the accelStepper instance. It just seemed simpler to set acceleration to a high enough value that acceleration would be complete on the first step. That's why 99999.